pkg/informer/informer.go

```diff
@@ -44,6 +75,162 @@ func InformerForResource(res schema.GroupVersionResource) (GenericInformer, erro
 	return i, nil
 }
 
+func isResourceAllowed(cm *v1.ConfigMap, group, kind string, allowedList []Resource, deniedList []Resource) bool {
+
+	var boolVar bool
+
+	// Ignore clusters and clusterstatus resources because these are handled by the aggregator.
+	// Ignore oauthaccesstoken resources because those cause too much noise on OpenShift clusters.
+	// Ignore projects as namespaces are overwritten to be projects on Openshift clusters - they tend to share
+	// the same uid.
+	list := []string{"events", "projects", "clusters", "clusterstatuses", "oauthaccesstokens"}
+
+	//remove all apiResources with kind in list
+	for _, name := range list {
+		if kind == name {
+			// klog.Info("list of names, resource kind:", name, kind)
+			return false
+		}
+	}
+
+	// remove denied resources from all apigroups if * otherwise remove from specific apigroups.
+	for _, deny := range deniedList {
+		for _, api := range deny.ApiGroups {
+			for _, rec := range deny.Resources {
+				if api == "*" && rec != "*" {
+					if rec == kind {
+						klog.Info("deny all group, specific resource", group, api, kind, rec)
+
+						return false
+					} else {
+						if api != "*" && rec == "*" {
+							if group == api {
+								klog.Info("deny specific group all resources", group, api, kind, rec)
+								return false
+							}
+						}
+
+					}
+				} else {
+					// if api != "*" && deny.Resources[i] != "*" {
+					if group == api && kind == rec {
+						klog.Info("deny specific group specific resources", group, api, kind, rec)
+						return false
+					}
+					// }
+				}
+			}
+		}
+	}
+
+	// if allowedlist is empty allow all resources, otherwise if * allow all groups specific
+	// resources if not * allow specific resources to specific groups:
+	if len(allowedList) == 0 {
+		return true
+	} else {
+		// klog.Info("In-coming ApiResource", kind)
+		for _, allow := range allowedList {
+			for _, api := range allow.ApiGroups {
+				for _, rec := range allow.Resources {
+					if api == "*" && rec != "*" { //all apigroups & specific resources
+						if rec == kind {
+							klog.Info("allow all group specific resources:", group, api, kind, rec)
+							return true
+
+						} else {
+							if api != "*" && rec == "*" { // specific apigroups & all resources
+								if group == api {
+									klog.Info("allow specific groups all resources:", group, api, kind, rec)
+									return true
+								}
+							}
+						}
+
+					} else {
+						// if api != "*" && allow.Resources[i] != "*" { //specific apigroups and resources
+						if group == api && kind == rec {
+							klog.Info("allow specific group specific resources", group, api, kind, rec)
+							return true
+						}
+						// }
+					}
+				}
+			}
+		}
+	}
+
+	return boolVar
```

Comment:
- You don't need `boolVar`, just `return true` here.
- Let's add a log.Warning() before this return. Normally we should match one of the rules above and exit before this point, so logging a warning can help us notice if we missed any case.